src/harness/harnessIO.ts

```diff
@@ -87,19 +87,20 @@ function createNodeIO(): IO {
     function listFiles(path: string, spec: RegExp, options: { recursive?: boolean; } = {}) {
         function filesInFolder(folder: string): string[] {
             let paths: string[] = [];
+            const dir = fs.opendirSync(folder);

```

Comment:
Could we just use `readdirSync` with `withFileTypes: true` instead? With recursion, we'd be holding directories open the entire way down, but maybe we never go deep enough.

(I'd also say that readdir has `recursive` now, but it's only available in Node 18+)

Reply:
That is a fair point. I don't think the we have a very deep directory structure in tests so it's probably fine.  I will read the whole current directory first close it and then continue reading sub directories just to be safe.

Reply:
Is withFileTypes enough to get all of the needed info? That's effectively what loading it all at once does.

I'm not against holding them open really, the code is plenty clear and I'm sure fine as is; ignore my hold open comment (speaking aloud too much)

Reply:
I changed it over to use withFileTypes